File: 240409/상품 코드/product-code.py
product, code = input().split()

class PRODUCT:
    def __init__(self, prod = "codetree", code = 50):
        self.prod = prod
        self.code = code

    def print_info(self):
        print("product", self.code, "is " + self.prod)

product_inst = PRODUCT()
product_inst.print_info()
product_inst = PRODUCT(product, code)
product_inst.print_info()

File: 260101/상품 코드/product-code.cpp
#include <iostream>
#include <string>

using namespace std;

class PRODUCT 
{
public:
    string prod;
    string code;

    PRODUCT(string p = "codetree", string c = "50")
        : prod(p), code(c) {}

    void print_info() 
    {
        cout << "product " << code << " is " << prod << endl;
    }
};

int main() 
{
    string product, code;
    cin >> product >> code;

    PRODUCT product_inst;
    product_inst.print_info();

    product_inst = PRODUCT(product, code);
    product_inst.print_info();

    return 0;
}
